cmd/apps/vault_app.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) arkade author(s) 2022. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallVault() *cobra.Command {
+	var vault = &cobra.Command{
+		Use:          "vault",
+		Short:        "Install vault",
+		Long:         `Install vault`,
+		Example:      `  arkade install vault`,
+		SilenceUsage: true,
+	}
+
+	vault.Flags().StringP("namespace", "n", "vault", "The namespace to install the chart")
+	vault.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set image.tag=1.11.2)")
+
```

Comment:
I would definitely map the followin' configurations, since afaik they are commonly used:
- Injector enabled: https://github.com/hashicorp/vault-helm/blob/main/values.yaml#L38
- Ingress enabled: https://github.com/hashicorp/vault-helm/blob/main/values.yaml#L360
(Normally I would let this pass without this but recently we had 2 issues about ingress integration for some charts.)

Reply:
> Please see the comments, nice work anyways!

Thanks a lot for the feedback. I decided not to overcomplicate PR and do a bare-minimum let's say.

Reply:
> I would definitely map the followin' configurations, since afaik they are commonly used:
> 
> * Injector enabled: https://github.com/hashicorp/vault-helm/blob/main/values.yaml#L38
> * Ingress enabled: https://github.com/hashicorp/vault-helm/blob/main/values.yaml#L360
>   (Normally I would let this pass without this but recently we had 2 issues about ingress integration for some charts.)

Injector is enabled by default. 
```
NAME                                   READY   STATUS    RESTARTS   AGE
vault-0                                1/1     Running   0          17h
vault-agent-injector-dbc799985-2sdzg   1/1     Running   0          17h
```

About Ingress - I'll check

Reply:
The comment is not about enabling the sidecart, but provide a flag that give users the ability to easily switch it on or off, also this is should be disabled by default, like it's stated in the contributing documentation:
https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md#recommended-settings-for-apps-for-arkade-install

Reply:
<img width="1552" alt="Screenshot 2022-08-16 at 10 49 59" src="https://user-images.githubusercontent.com/20466436/184838848-c7338529-2e99-46b9-b98d-84b2021b0a99.png">

ingress works, tested on minikube:
```
NAME    CLASS   HOSTS                 ADDRESS        PORTS   AGE
vault   nginx   chart-example.local   192.168.49.2   80      71s
```

Should I add configuration for some other params like domain/tls? They can be for sure added via `--set`

Reply:
Disabled persistence by default, made it configurable via `--persistence`

Reply:
I think using `--set` for domain and tls is fine by me, but you can add those if you think it's necessary.

Reply:
I'll add domain but TLS better to be configured via --set, because it can involve different options. Especially, for fully working vault's tls.

Reply:
@Shikachuu Added flag to configure domain

```
$ ./arkade install vault --ingress=true --domain=vault-example.com

$ kgi
vault   nginx   vault-example.com   192.168.49.2   80      3m32s
```

Let me know if you think it will be useful to print url after chart installation to the cli output. I was thinking about something like that. But I'm not sure if it's common practice for arkade. 
```
if ingressEnabled && ingressDomain != "" {
    vaultAccessMsg := "Vault is available via" + ingressDomain
}
```

Reply:
Can you add a check in the `PreRunE` that `--domain` flag is only available when using the `--ingress` flag please?

It would be a nice touch to add a dynamic message about ingress but I don't think it's necessarry, I mean it should be obvious when you used the `--ingress` and `--domain` flags.

Reply:
Added. Follow up question: should I add verbose errors for errors in `PreRunE`, e.g. https://github.com/alexellis/arkade/blob/master/cmd/apps/cassandra_app.go#L60

Reply:
I don't think it's necessary, I think cobra errors are a-okay.